core/src/main/java/io/grpc/internal/ManagedChannelImpl.java

```diff
@@ -984,7 +994,7 @@ void remove(RetriableStream<?> retriableStream) {
     // Must be called from syncContext
     private void handleInternalSubchannelState(ConnectivityStateInfo newState) {
       if (newState.getState() == TRANSIENT_FAILURE || newState.getState() == IDLE) {
-        nr.refresh();
```

Comment:
Move `nr` to `NameResolverListenerImpl`?

Reply:
Done.